docs/.storybook/preview.tsx

```diff
@@ -110,32 +107,123 @@ const preview = {
-          "Guides",
-          [
-            "App starter",
-            "Layout and spacing",
-            "Tailwind",
-            [
+      storySort: (a, b) => {
+        // Two exact same stories
+        if (a.id === b.id) {
+          return 0
+        }
+
+        // Custom ordering for Tailwind docs
+        if (a.id.includes("tailwind") && b.id.includes("tailwind")) {
+          if (b.title.includes("References")) {
+            if (a.title.includes("References")) {
+              // Overview above all other stories
+              if (a.title.includes("Overview")) return -1
+              if (b.title.includes("Overview")) return 1
+              return 0
+            }
+            // Put References below other stories
+            return -1
+          }
+
+          // Put References below other stories
+          if (a.title.includes("References")) return 1
+
+          if (a.type === "docs" && b.type === "docs") {
+            const docs = [
               "Overview",
               "Getting Started",
               "Configuration",
               "Working with Tailwind",
-              "*",
-              "Utility Class References",
-              ["Overview", "*"],
-            ],
-          ],
-          // This will group new folders like Actions, Overlays, etc in the same way
-          "*",
-          ["*", ["*", ["Usage Guidelines", "API Specification", "*"]]],
+            ]
+            const docsNameA = a.title.split("/").pop()
+            const docsNameB = b.title.split("/").pop()
+            if (docs.includes(docsNameA) && docs.includes(docsNameB)) {
+              const docsDifference =
+                docs.indexOf(docsNameA) - docs.indexOf(docsNameB)
+              if (docsDifference !== 0) {
+                // Sort docs by listed order
+                return docsDifference
+              }
+            }
+
+            // Sort listed docs above unlisted ones
+            if (docs.includes(docsNameA)) return -1
+            // Sort unlisted docs below listed ones
+            if (docs.includes(docsNameB)) return 1
+          }
+
+          // All other stories appear in the order they are defined
+          return 0
+        }
+
+        const customDocNames = ["Usage Guidelines", "API Specification"]
+        // Don't type the param - we can't use TypeScript within storySort
+        const removeCustomDocNames = title => {
+          return customDocNames.reduce((acc, docName) => {
+            const regex = new RegExp(`/${docName}$`)
+            return acc.replace(regex, "")
+          }, title)
+        }
+
+        const titleA = removeCustomDocNames(a.title)
+        const titleB = removeCustomDocNames(b.title)
+
+        const groupA = titleA.split("/")[0]
+        const groupB = titleB.split("/")[0]
+
+        const groups = [
+          "Introduction",
+          "Guides",
+          "Actions",
+          "Containers",
+          "Illustrations",
+          "Layout",
+          "Overlays",
```

Comment:
May as well put all the groups planned here

Reply:
I didn't want to put them in until we actually use them 😬 It would be easy to add anyway when the time comes